src/molecules/Snackbar/SnackbarProvider.native.js

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useCallback, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Snackbar from './Snackbar';
+import SnackbarContext from './SnackbarContext';
+
+const DEFAULT_DISMISS_DURATION = 3000;
+
+const SnackbarProvider = ({ children }) => {
+  const [snackbarProps, setSnackbarProps] = useState({
+    variant: undefined,
+    text: '',
+    actionText: undefined,
+    onAction: undefined,
+    showDismissButton: undefined,
+    onDismiss: undefined,
+    maxLines: undefined,
+    position: { top: undefined, bottom: 0, left: 0, right: 0 },
+  });
+  const [isVisible, setIsVisible] = useState(false);
+  const timerRef = useRef(null);
+  const dismissAfterDuration = useCallback((duration = DEFAULT_DISMISS_DURATION) => {
+    if (timerRef.current) {
+      clearTimeout(timerRef.current); // clear existing timer
+    }
+    timerRef.current = setTimeout(() => setIsVisible(false), duration);
+  }, []);
+
+  const dismiss = useCallback(() => {
+    if (timerRef.current) {
+      clearTimeout(timerRef.current); // clear existing timer
+    }
+    if (isVisible) {
+      setIsVisible(false);
+    }
+  }, [isVisible]);
+
+  const invokeNewSnackBar = useCallback(
+    (config) => {
+      setSnackbarProps(config);
```

Comment:
Why do you need another setState for snackbar? the props you need to pass to snackbar cab just be plain object and when you toggle isVisible anyways it'll toggle the behavior

Reply:
Snackbar is already rendered at the top of the app, it's only hidden using state. Now we need to change the state so that we can make Snackbar visible & update it's props

Reply:
You mean you'll always render the snackbar in the DOM? If yes then it's incorrect. if the snackbar is hidden then it should not be rendered at all the provider should return `null` basically in that case

Reply:
If I return null when the user dismisses it, then it won't have that hide animation. 

Does this mean that I have to maintain another boolean state similar to `isVisible` that will return the snackbar or null and will be turned to false when exit animation is complete?